### DISCUSSION TITLE
QTable with editable content

### DISCUSSION THREAD

**Discussion by s-light** [quasarframework/quasar#14772] [General - Components / Directives / etc]
Hello Community,

i have an concept for a *spreadsheet-like* data importer.

for this i found [QTable](https://quasar.dev/vue-components/table#responsive-tables) could be a good starting point -
but i will need *inline* editing options.
what is the best approach here?
is it a good idea to use the [Body-cell slot](https://quasar.dev/vue-components/table#example--body-cell-slot) with an [input textfield](https://quasar.dev/vue-components/input) ?
or is there some other nicer way of doing this?

i think i would need some *heavy* keyboard checks as i want to enable keyboard-navigation from cell to cell...
(spreadsheet-style)

second i also have to implement copy & paste functionality -
(later maybe csv/excel/libreoffice spreadsheet drop area too...)
for the paste functionality: 
i don*t know what columns i will get - 
and maybe no column names at all -
my idea is to add some sort of *drag & drop* option for predefined column labels to the found *nameless* columns...
maybe anybody has some thoughts on this - or ideas where to start looking / tips.......

looking forward to your ideas and thoughts :-)

sunny greetings
stefan

**Comment by s-light** [comment]
just found [QModelTd](https://www.npmjs.com/package/quasar-app-extension-qmodeltd)
will test if this is working for me ;-)

**Reply by s-light** [reply]
i did not tested QModelTd.
instead i just tried my luck and got it working :-)
good reference was question  #14922.

```vue
<q-table
    title="My Import"
    dense
    :rows="rows"
    :columns="columns"
    row-key="name"
    :hide-pagination="true"
    myInitialPagination="0"
    :pagination="{ rowsPerPage: 0 }"
>
    <template v-slot:body="props">
        <q-tr :props="props">
            <q-td
                v-for="col in props.cols"
                :key="col.name"
                :props="props"
            >
                <q-input
                    v-model="props.row[col.name]"
                    dense
                    borderless
                    autofocus
                ></q-input>
            </q-td>
        </q-tr>
    </template>
</q-table>
```

**Comment by hekep** [comment]
QTable with `v-slot:body="props"` is "OK" starting point, whoever editing fields later has limiting event handling (?). 

I am looking for solution to this:
https://github.com/quasarframework/quasar/discussions/14809
